src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -948,18 +947,32 @@ bool LibraryCallKit::inline_vector_mem_masked_operation(bool is_store) {
     }
     return false; // should be primitive type
   }
-
   BasicType elem_bt = elem_type->basic_type();
   int num_elem = vlen->get_con();
+
   int sopc = is_store ? Op_StoreVectorMasked : Op_LoadVectorMasked;
-  if (!arch_supports_vector(sopc, num_elem, elem_bt, VecMaskUseLoad) ||
-      !Matcher::match_rule_supported_vector_masked(sopc, num_elem, elem_bt)) {
+  bool use_predicate = Matcher::has_predicated_vectors() &&
+                       arch_supports_vector(sopc, num_elem, elem_bt, VecMaskUseLoad) &&
+                       Matcher::match_rule_supported_vector_masked(sopc, num_elem, elem_bt);
```

Comment:
Can we push both  Matcher::has_predicated_vectors() && Matcher::match_rule_supported_vector_masked() into arch_supports_vector and either add an explicit bool argument to it or a new enum value to be passed i.e. VecMaskUseLoad | VecPredOper  under which these checks are enforced.

Reply:
Good idea! Will do this later. Thanks so much!